File: backend/src/main/java/com/kostone/marble/domain/activity/ActivityLog.java
package com.kostone.marble.domain.activity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "activity_log")
@Getter
@Setter
@NoArgsConstructor
public class ActivityLog {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Enumerated(EnumType.STRING)
    @Column(name = "entity_type", nullable = false, length = 30)
    private ActivityEntityType entityType;

    @Column(name = "entity_id", nullable = false)
    private UUID entityId;

    @Enumerated(EnumType.STRING)
    @Column(name = "action", nullable = false, length = 30)
    private ActivityAction action;

    @Column(name = "performed_by_user_id")
    private UUID performedByUserId;

    @Column(name = "performed_by_name", nullable = false, length = 255)
    private String performedByName;

    @Column(name = "customer_name", length = 255)
    private String customerName;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "reason", columnDefinition = "TEXT")
    private String reason;

    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt = OffsetDateTime.now();
}

File: frontend/src/api/activityLog.ts
import axios from 'axios'

export type ActivityAction = 'ORDER_CREATED' | 'ORDER_STATUS_CHANGED' | 'ORDER_DELETED' | 'ORDER_RESTORED'

export const ACTIVITY_ACTION_HE: Record<ActivityAction, string> = {
  ORDER_CREATED: 'הזמנה נוצרה',
  ORDER_STATUS_CHANGED: 'סטטוס הזמנה עודכן',
  ORDER_DELETED: 'הזמנה נמחקה',
  ORDER_RESTORED: 'הזמנה שוחזרה',
}

export interface ActivityLogEntry {
  id: string
  entityType: string
  entityId: string
  action: ActivityAction
  performedByName: string | null
  customerName: string | null
  description: string | null
  reason: string | null
  createdAt: string
}

export async function fetchActivityLog(): Promise<ActivityLogEntry[]> {
  const { data } = await axios.get<ActivityLogEntry[]>('/api/v1/activity-log')
  return data
}
